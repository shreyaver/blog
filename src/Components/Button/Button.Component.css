.Button {
    margin: 40px auto 20px;
    border: 1px solid rgb(216, 216, 216);
    width: 5%;
    min-width: 150px;
    height: 40px;
    color:grey;
    font-size: 15px;
    padding: 10px 10px;
}