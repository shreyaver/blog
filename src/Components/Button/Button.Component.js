import React from 'react';
import './Button.Component.css';

const Button = (props) => {
  return (
    <button type={props.type} className="Button" onClick={props.onClick}>
      {props.caption}
    </button>
  );
}
export default Button;