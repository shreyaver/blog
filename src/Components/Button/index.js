import React from 'react';
import './Button.css';

const Button = (props) => {
  return (
    <button type={props.type} className="Button" onClick={props.onClick}>
      {props.caption}
    </button>
  );
}
export default Button;