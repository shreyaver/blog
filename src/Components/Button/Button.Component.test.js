import React from 'react';
import renderer from 'react-test-renderer';
import Button from './Button.Component';
import { shallow } from 'enzyme'

describe('Button', () => {
  it('renders without crashing', () => {
    
    const button = renderer.create(<Button caption = 'CREATE POSTS' />).toJSON();
    expect(button).toMatchSnapshot();
  });
  it('sets button caption to passed value', () => {
    
    const wrapper = shallow(<Button caption = 'CREATE POSTS' onClick={jest.fn()}/>);
    expect(wrapper.find('.Button').text()).toEqual('CREATE POSTS');
  });
  it('calls passed function on click', () => {
    const mockFunction = jest.fn();
    const wrapper = shallow(<Button caption = 'CREATE POSTS' onClick={mockFunction}/>);
    wrapper.find('.Button').simulate('click');
    expect(mockFunction).toHaveBeenCalled();
  });
});
