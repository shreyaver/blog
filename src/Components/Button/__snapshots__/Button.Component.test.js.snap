// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button renders without crashing 1`] = `
<button
  className="Button"
>
  CREATE POSTS
</button>
`;
